### ISSUE TITLE
PHP 7.4: Necessary null distribution

### ISSUE THREAD

**Issue by Hill-98** [kalessil/phpinspectionsea#1455]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended) 4.0.1.1 |
| Language level | PHP 7.4                       |

```php
class Logger
{
    private static ?Logger $logger;

    public static function init(): void
    {
        if (self::$logger !== null) {
            return;
        }
    }
}

Logger::init();
```
If you do not assign null in PHP 7.4, the above code error: `$logger must not be accessed before initialization`

Now, PHP inspections think null is unnecessary

**Comment by rentalhost** [comment]
The problem is: how identify that `$logger` was set or not? And where it MUST be set? I mean, maybe I set it latter in another scope, for public properties for instance.

**Comment by Hill-98** [comment]
This may be a bit cumbersome, but if there are null type declaration, disable detection, feasible?

**Comment by rentalhost** [comment]
I don't think I could understand exactly the case here. May I ask you to try to explain some more?

**Comment by Hill-98** [comment]
If the property exists type declaration (eg: ?Class), assign null, it does not detect if necessary.
If the assignment is not null, only use empty () for testing.

These may not be a good idea

**Comment by kalessil** [comment]
Fixed!

**Comment by orklah** [comment]
Cool!
